### PR TITLE
GUVNOR-2318: Need settings to disable background build when opening Project Authoring

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-2318

### CODE REVIEW

**Comment by manstis** [comment]
See https://github.com/droolsjbpm/kie-docs/pull/20 too.

**Comment by csadilek** [comment]
+1

**Comment by csadilek** [comment]
master: 10e3647d143f4405f04226d2d595c46f5f5e85e0

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/test/java/org/kie/workbench/common/screens/explorer/client/widgets/BaseViewPresenterTest.java [hunk 1]
```diff
@@ -255,6 +256,40 @@ public void testAutomaticProjectBuildEnabled() {
...
+
+            verify( buildServiceActual,
+                    never() ).build( any( Project.class ) );
+
+        } finally {
+            if ( spBuildDisableProjectExplorer != null ) {
+                System.setProperty( "build.disable-project-explorer",
+                                    spBuildDisableProjectExplorer );
+            }
+        }
```
Comment (yurloc) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/test/java/org/kie/workbench/common/screens/explorer/client/widgets/BaseViewPresenterTest.java]: @manstis I assume the purpose of this `finally` block is to restore original system properties state. If that is correct, the string `"build.disable-project-explorer"` looks like typo and should be replaced with the `BUILD_PROJECT_PROPERTY_NAME` constant which holds a different string value.

Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/test/java/org/kie/workbench/common/screens/explorer/client/widgets/BaseViewPresenterTest.java]: Good spot @yurloc 

See https://github.com/droolsjbpm/kie-wb-common/pull/196

**Pull request closed without merging**